src/test/java/org/opensearch/security/dlic/rest/api/TenantsActionTest.java

```diff
@@ -0,0 +1,137 @@
+                    break;
+            }
+        }
+
+        String[] caseTwoRequests = { ENDPOINT + "/tenant1", ENDPOINT + "/tenant1", ENDPOINT + "/tenant1",
+                ENDPOINT + "/tenant1", ENDPOINT + "/tenant1" };
+        String[] caseTwoBodies = { "{\"description\":\"create tenant 1\"}", "{\"description\":\"create tenant 2\"}",
+                "{\"description\":\"create tenant 3\"}", "{\"description\":\"create tenant 4\"}",
+                "{\"description\":\"create tenant 5\"}" };
+        HttpResponse[] caseTwoResponses = executeMultipleAsyncPutRequest(caseTwoRequests, caseTwoBodies);
+        created = false;
+        for (HttpResponse response : caseTwoResponses) {
+            int sc = response.getStatusCode();
+            switch (sc) {
+                case HttpStatus.SC_CREATED:
+                    Assert.assertFalse(created);
+                    created = true;
+                    break;
+                case HttpStatus.SC_OK:
+                    break;
+                default:
+                    Assert.assertEquals(HttpStatus.SC_CONFLICT, sc);
+                    break;
+            }
+        }
+
+        String[] caseThreeRequests = { ENDPOINT + "/tenant1", ENDPOINT + "/tenant1", ENDPOINT + "/tenant1",
+                ENDPOINT + "/tenant1", ENDPOINT + "/tenant1", ENDPOINT + "/tenant1", ENDPOINT + "/tenant1",
+                ENDPOINT + "/tenant1", ENDPOINT + "/tenant1", ENDPOINT + "/tenant1" };
+        String[] caseThreeBodies = { "{\"description\":\"create tenant 1\"}", "{\"description\":\"create tenant 2\"}",
+                "{\"description\":\"create tenant 3\"}", "{\"description\":\"create tenant 4\"}",
+                "{\"description\":\"create tenant 5\"}", "{\"description\":\"create tenant 6\"}",
+                "{\"description\":\"create tenant 7\"}", "{\"description\":\"create tenant 8\"}",
+                "{\"description\":\"create tenant 9\"}", "{\"description\":\"create tenant 10\"}" };
+        HttpResponse[] caseThreeResponses = executeMultipleAsyncPutRequest(caseThreeRequests, caseThreeBodies);
+        created = false;
+        for (HttpResponse response : caseThreeResponses) {
+            int sc = response.getStatusCode();
+            switch (sc) {
+                case HttpStatus.SC_CREATED:
+                    Assert.assertFalse(created);
+                    created = true;
+                    break;
+                case HttpStatus.SC_OK:
+                    break;
+                default:
+                    Assert.assertEquals(HttpStatus.SC_CONFLICT, sc);
+                    break;
+            }
+        }
+    }
+
+    private HttpResponse[] executeMultipleAsyncPutRequest(final String[] requests,
+            final String[] bodies) throws Exception {
+        final int numOfRequests = Math.min(requests.length, bodies.length);
+        final ExecutorService executorService = Executors.newFixedThreadPool(numOfRequests);
```

Comment:
None of the async request logic is needed, after the first `PUT` all future requests will fail with conflict because the tenant has already been created.  You can verify this by setting the threadpool size to 1.

Reply:
Do you mean even if they're not synchronous? This is interesting because when running `curl -X PUT https://localhost:9200/_plugins/_security/api/tenants/tenant_1 -u 'admin:admin' --insecure -d '{"description": "create tenant 1"}' -H 'Content-Type: application/json'` in my terminal twice in a row I get 20X responses, with the first one giving a "created" response, and the second one giving a "updated" response.

Reply:
To be honest, I'm not sure - I just changed the setting associated with this line to `1` and noticed that nothing failed - which would indicate the tests aren't verifying something correctly if you'd expect them to fail.  Its possible there is a bug in that isn't causing the failure state to be noticed in the code, maybe the issue is because cause of how that validate loop works:

```
                case HttpStatus.SC_OK:
                    break;
```

However, in general test around 'racing' for a conflict at an API level are going to be flaky typically, so I would generally discourage test of this nature unless there is a strongly deterministic way to get a reproduction, such as being able to control the timing across threads, or use C/ETags to ensure processing order.